src/category_theory/monoidal/Mon_.lean

```diff
@@ -273,6 +299,174 @@ def equiv_lax_monoidal_functor_punit : lax_monoidal_functor (discrete punit.{u+1
+together form a strength that equips the tensor product functor with a monoidal structure, and the
+monoid axioms for the tensor product follow from the monoid axioms for the tensor factors plus the
+properties of the strength (i.e., monoidal functor axioms).  The strength `tensor_μ` of the tensor
+product functor has been defined in `category_theory.monoidal.braided`.  Its properties, stated as
+independent lemmas in that module, are used extensively in the proofs below.
+
+To complete the definition of the monoidal category structure on the category of monoids, we need
+to provide definitions of associator and unitors.  The obvious candidates are the associator and
+unitors from `C`, but we need to prove that they are monoid morphisms, i.e., compatible with unit
+and multiplication.  These properties translate to the monoidality of the associator and unitors
+(with respect to the monoidal structures on the functors they relate), which have also been proved
+in `category_theory.monoidal.braided`.
+
+-/
+
+variables {C} [braided_category C]
+
+lemma Mon_tensor_one_mul (M N : Mon_ C) :
+    ((λ_ (𝟙_ C)).inv ≫ (M.one ⊗ N.one) ⊗ 𝟙 (M.X ⊗ N.X)) ≫
+    tensor_μ C (M.X, N.X) (M.X, N.X) ≫ (M.mul ⊗ N.mul)
+  = (λ_ (M.X ⊗ N.X)).hom :=
+begin
+  rw [←category.id_comp (𝟙 (M.X ⊗ N.X)), tensor_comp],
+  slice_lhs 2 3 { rw [←tensor_id, tensor_μ_natural] },
+  slice_lhs 3 4 { rw [←tensor_comp, one_mul M, one_mul N] },
+  symmetry,
+  exact tensor_left_unitality C M.X N.X,
+end
+
+lemma Mon_tensor_mul_one (M N : Mon_ C) :
+    (𝟙 (M.X ⊗ N.X) ⊗ (λ_ (𝟙_ C)).inv ≫ (M.one ⊗ N.one)) ≫
+    tensor_μ C (M.X, N.X) (M.X, N.X) ≫ (M.mul ⊗ N.mul)
+  = (ρ_ (M.X ⊗ N.X)).hom :=
+begin
+  rw [←category.id_comp (𝟙 (M.X ⊗ N.X)), tensor_comp],
+  slice_lhs 2 3 { rw [←tensor_id, tensor_μ_natural] },
+  slice_lhs 3 4 { rw [←tensor_comp, mul_one M, mul_one N] },
+  symmetry,
+  exact tensor_right_unitality C M.X N.X,
+end
+
+lemma Mon_tensor_mul_assoc (M N : Mon_ C) :
+    (tensor_μ C (M.X, N.X) (M.X, N.X) ≫ (M.mul ⊗ N.mul) ⊗ 𝟙 (M.X ⊗ N.X)) ≫
+    tensor_μ C (M.X, N.X) (M.X, N.X) ≫
+    (M.mul ⊗ N.mul)
+  = (α_ (M.X ⊗ N.X) (M.X ⊗ N.X) (M.X ⊗ N.X)).hom ≫
+    (𝟙 (M.X ⊗ N.X) ⊗ tensor_μ C (M.X, N.X) (M.X, N.X) ≫ (M.mul ⊗ N.mul)) ≫
+    tensor_μ C (M.X, N.X) (M.X, N.X) ≫
+    (M.mul ⊗ N.mul) :=
+begin
+  rw [←category.id_comp (𝟙 (M.X ⊗ N.X)), tensor_comp],
+  slice_lhs 2 3 { rw [←tensor_id, tensor_μ_natural] },
+  slice_lhs 3 4 { rw [←tensor_comp, mul_assoc M, mul_assoc N, tensor_comp, tensor_comp] },
+  slice_lhs 1 3 { rw [tensor_associativity] },
+  slice_lhs 3 4 { rw [←tensor_μ_natural] },
+  slice_lhs 2 3 { rw [←tensor_comp, tensor_id] },
+  simp only [category.assoc],
+end
+
+lemma one_associator {C : Type u₁} [category.{v₁} C] [monoidal_category.{v₁} C] {M N P : Mon_ C} :
+    ((λ_ (𝟙_ C)).inv ≫ ((λ_ (𝟙_ C)).inv ≫ (M.one ⊗ N.one) ⊗ P.one)) ≫ (α_ M.X N.X P.X).hom
+  = (λ_ (𝟙_ C)).inv ≫ (M.one ⊗ (λ_ (𝟙_ C)).inv ≫ (N.one ⊗ P.one)) :=
```

Comment:
It may make sense to move all the theorems which don't require a braiding to a preliminary section. What do you think?